### PR TITLE
Task [1] - Update burger state and desciption in swipe

### PR DESCRIPTION
**Pull request by Yaelsarusi**

### CODE REVIEW

**Review by daphnalev — APPROVED** [review]

File: app/src/main/java/com/example/graburger/BurgerItemModel.java [hunk 1]
```diff
@@ -3,9 +3,9 @@
 import java.io.Serializable;
 
 enum BunType {
-    REGULAR(0, R.drawable.upper_bun_regular, R.drawable.lower_bun_regular,"Regular wheat bun"),
-    GLUTEN_FREE(1, R.drawable.upper_bun_gluten_free, R.drawable.lower_bun_gluten_free,"Gluten free bun"),
-    WHOLE(2, R.drawable.upper_bun_whole, R.drawable.lower_bun_whole, "Whole wheat bun");
+    REGULAR(0, R.drawable.upper_bun_regular, R.drawable.lower_bun_regular,"regular wheat bun"),
```
Comment (shakedweitz) [inline app/src/main/java/com/example/graburger/BurgerItemModel.java:6]: We should localize all the strings

File: app/src/main/java/com/example/graburger/BurgerItemModel.java [hunk 2]
```diff
@@ -28,14 +28,16 @@ public int getImage(Boolean isUpper ) {
         this.title = title;
     }
 
-    static BunType[] all(){ return new BunType[] {REGULAR, GLUTEN_FREE, WHOLE};}
+    static BunType[] all(){ return BunType.values();}
+
+    static BunType get(int i){return all()[i];}
```
Comment (shakedweitz) [inline app/src/main/java/com/example/graburger/BurgerItemModel.java:33]: consider renaming "i" to "index"

File: app/src/main/java/com/example/graburger/BurgerItemModel.java [hunk 3]
```diff
@@ -192,20 +205,19 @@ public void updateImage(){
     }
 
     public void updateDesc(){
-        super.setDesc(bunType.getTitle() + bunType.getTitle() +
-                cheeseType.getTitle() + pattyType.getTitle() +
-                tomatoType.getTitle() + lettuceType.getTitle() +
-                sauceType.getTitle());
+        super.setDesc(" " + bunType.getTitle() + ", " + cheeseType.getTitle() + ", " +
+                pattyType.getTitle() + ", " + tomatoType.getTitle() + ", " +
+                lettuceType.getTitle() + ", " + sauceType.getTitle());
```
Comment (shakedweitz) [inline app/src/main/java/com/example/graburger/BurgerItemModel.java:210]: I think using string.format could be more elegant

File: app/src/main/java/com/example/graburger/BurgerItemModel.java [hunk 4]
```diff
@@ -192,20 +205,19 @@ public void updateImage(){
...
     }
 
     private void updateBurger(){
         updateImage();
         updateDesc();
     }
 
-    public void updateBurger(BunType newBun){ this.bunType = newBun; updateBurger();}
+    public void updateBurger(BunType newBun){ this.bunType = newBun; updateBurger(); }
```
Comment (shakedweitz) [inline app/src/main/java/com/example/graburger/BurgerItemModel.java:218]: nit: new line after opening brace

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -2,4 +2,7 @@
     <string name="app_name">Graburger</string>
     <string name="checkout">Checkout</string>
     <string name="my_order">My Order</string>
+    <string name="CurOrder">Current order: </string>
```
Comment (shakedweitz) [inline app/src/main/res/values/strings.xml:5]: we should verify the coding style for the name and be consistent

File: app/src/main/java/com/example/graburger/buildBurgerActivity.java [hunk 6]
```diff
@@ -53,47 +57,120 @@ public void onPageScrollStateChanged(int state) {}
             cheese.add(new FoodItemModel(c.getImage(), c.getTitle()));
         }
 
-        setViewPager(cheese, R.id.Cheese, curBurger.getCheese().getPosition());
+        setViewPager(cheese, R.id.Cheese, curBurger.getCheese().getPosition(),
+                new ViewPager.OnPageChangeListener() {
```
Comment (shakedweitz) [inline app/src/main/java/com/example/graburger/buildBurgerActivity.java:61]: we should check listener options, maybe there is a way with less code duplication

File: app/src/main/java/com/example/graburger/BurgerItemModel.java (hunk 2 above)
Comment (Yaelsarusi) [reply app/src/main/java/com/example/graburger/BurgerItemModel.java:33]: you're right, changed it in the next pull request I'll send.

File: app/src/main/java/com/example/graburger/buildBurgerActivity.java (hunk 6 above)
Comment (Yaelsarusi) [reply app/src/main/java/com/example/graburger/buildBurgerActivity.java:61]: I've tidied it up (In the next pull request), but to remove the redundant duplication, I think we will need to create a function that will receive a Runnable, and return a ViewPager.OnPageChangeListener(), that calls the Runnable() and recieves the item that needs to update. For that we will need to change all the Enums to Singelton Classes, and create a factory to get them. They will all extend a BaseType class. 
It's prettier, however it's allot of work. So lets discuss this.

**Pull request merged**